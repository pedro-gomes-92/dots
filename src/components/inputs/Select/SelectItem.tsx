import React, { FunctionComponent } from 'react';
import classnames from 'classnames';
import { BaseProps, Base } from '../../Base';

export interface SelectItemProps extends BaseProps {
  text: string;
  value?: FormValue;
}

export const SelectItem: FunctionComponent<SelectItemProps> = ({
  className,
  text,
  attribute,
  value,
  ...rest
}: SelectItemProps): JSX.Element => (
  <Base className={classnames('select-option', className)} attribute={{ ...attribute, value }} {...rest}>
    {text}
  </Base>
);

SelectItem.defaultProps = {
  tag: 'option',
};
