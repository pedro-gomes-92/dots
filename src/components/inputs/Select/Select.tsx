import React, { FunctionComponent } from 'react';
import classnames from 'classnames';
import { Children } from '../../typings';
import { SelectItem } from './SelectItem';
import { Base } from '../../Base';
import { BaseInput, BaseInputProps } from '../BaseInput';

export interface SelectProps extends BaseInputProps {
  children: Children<typeof SelectItem>;
}

export const Select: FunctionComponent<SelectProps> = ({
  className,
  style,
  children,
  ...rest
}: SelectProps): JSX.Element => (
  <Base className={classnames('select', className)} style={style}>
    <BaseInput {...rest}>{children}</BaseInput>
  </Base>
);

Select.defaultProps = {
  tag: 'select',
};
