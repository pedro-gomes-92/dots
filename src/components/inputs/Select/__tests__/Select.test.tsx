import React from 'react';
import { render } from '@testing-library/react';

import { Select } from '../Select';
import { SelectItem } from '../SelectItem';

describe('Given <Select />', (): void => {
  let container: HTMLElement;
  describe('When initializes', (): void => {
    it('Then should match the snapshot', (): void => {
      container = render(
        <Select>
          <SelectItem text="Testing 1" />
        </Select>,
      ).container;
      expect(container).toMatchSnapshot();
    });

    describe('And uses children', (): void => {
      it('Then should match the snapshot', (): void => {
        container = render(
          <Select>
            <SelectItem text="Testing 1" />
            <SelectItem text="Testing 2" />
          </Select>,
        ).container;
        expect(container).toMatchSnapshot();
      });
    });
  });
});
