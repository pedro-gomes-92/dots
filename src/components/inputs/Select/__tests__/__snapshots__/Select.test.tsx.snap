// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <Select /> When initializes And uses children Then should match the snapshot 1`] = `
<div>
  <div
    class="select"
  >
    <select
      class="input"
    >
      <option
        class="select-option"
      >
        Testing 1
      </option>
      <option
        class="select-option"
      >
        Testing 2
      </option>
    </select>
  </div>
</div>
`;

exports[`Given <Select /> When initializes Then should match the snapshot 1`] = `
<div>
  <div
    class="select"
  >
    <select
      class="input"
    >
      <option
        class="select-option"
      >
        Testing 1
      </option>
    </select>
  </div>
</div>
`;
